extern/CMakeProject-png.cmake

```diff
@@ -42,5 +46,9 @@ else()
     sm_add_compile_definition("png" _CRT_SECURE_NO_WARNINGS)
   endif()
 
-  target_include_directories("png" PUBLIC "zlib")
+  target_include_directories("png" PUBLIC
+    "zlib"
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/libpng>
+    $<INSTALL_INTERFACE:libpng>
+  )
```

Comment:
I am not sure, if this is actually needed. Let me check that.

Reply:
Ah, I remember. This makes sure, the libpng use the correct libpng headers with the vanilla libpng structure. See [target_include_directories — CMake Documentation](https://cmake.org/cmake/help/latest/command/target_include_directories.html)